### PR TITLE
NF: fix_permissions node 

### LINKED ISSUE #390: node: to fixup permissions of files
in many cases extracted files, if coming e.g. from windows, would have execution bit set.  We should clear it out.  Sure thing it could be an option for add_archive_content but imho should be a separate node allowing for specification of file patterns, mime types, etc to reset those executable bits (or may be wider range, e.g. setting up groups stickiness etc)

### PR DESCRIPTION
**Pull request by glalteva**
Closes #390

### CODE REVIEW

**Comment by glalteva** [comment]
@yarikoptic 

`lgr.debug('Changing permissions for file %s from %s to %s' % filename, per, nper)` gives the following error:

TypeError: not enough arguments for format string

I have 3 for 3- wasn't able to find much help online regarding this issue.
This line of code is currently commented out as to not fail any test.

**Comment by yarikoptic** [comment]
replace % with ,  OR place all arguments to % within a tuple (i.e. enclose within ()) .... see other examples of .debug invocations

File: datalad/crawler/nodes/misc.py [hunk 1]
```diff
@@ -29,6 +31,76 @@
...
+                if per[2] == '6' or '4':
+                    os.chmod(filename, st.st_mode | S_IXOTH)
+            else:
+                # strip everyone away from executing
+                current = stat.S_IMODE(os.lstat(filename).st_mode)
+                os.chmod(filename, current & ~stat.S_IEXEC)
+                current = stat.S_IMODE(os.lstat(filename).st_mode)
+                os.chmod(filename, current & ~stat.S_IXGRP)
+                current = stat.S_IMODE(os.lstat(filename).st_mode)
+                os.chmod(filename, current & ~stat.S_IXOTH)
```
Comment (yarikoptic) [inline datalad/crawler/nodes/misc.py]: please try to make the same logic without duplication -- You can loop through needed "values" and then use exactly the same code, without copy/pasting the same 2 lines.

File: datalad/crawler/nodes/misc.py [hunk 2]
```diff
@@ -29,6 +31,76 @@
...
+
+            nper = oct(os.stat(filename)[ST_MODE])[-3:]
+            lgr.debug('Changing permissions for file %s from %s to %s', filename, per, nper)
+
+        yield nper
```
Comment (yarikoptic) [inline datalad/crawler/nodes/misc.py]: nodes must yield "data", i.e. dictionary with fields... in this case I don't think that node should yield permissions at all, just fix, and yield original data.  In the test just assess them "manually" through os.stat

File: datalad/crawler/nodes/misc.py (hunk 1 above)
Comment (glalteva) [inline datalad/crawler/nodes/misc.py]: @yarikoptic 
I attempted to shorten this section with the following code:
`# strip everyone away from executing
                for scope in permissions:
                    current = stat.S_IMODE(os.lstat(filename).st_mode)
                    os.chmod(filename, current & ~stat.scope)`

However, `scope` is searched for in stat and not referenced from the for-loop. I tried concatting the variable first and typecasting, but have had no success. Any suggestions?

File: datalad/crawler/nodes/misc.py [hunk 3]
```diff
@@ -12,8 +12,10 @@
 import inspect
 import os
 import re
+import stat
 
-from os.path import curdir, join as opj, split as ops
+from stat import *
+from os.path import curdir,isdir, isabs, exists, join as opj, split as ops
```
Comment (yarikoptic) [inline datalad/crawler/nodes/misc.py]: feels like you are missing some space ;)

File: datalad/crawler/nodes/misc.py [hunk 4]
```diff
@@ -12,8 +12,10 @@
 import inspect
 import os
 import re
+import stat
 
-from os.path import curdir, join as opj, split as ops
+from stat import *
```
Comment (yarikoptic) [inline datalad/crawler/nodes/misc.py]: in general `import *` is discouraged, please list each imported entity explicitly

File: datalad/crawler/nodes/misc.py (hunk 1 above)
Comment (yarikoptic) [inline datalad/crawler/nodes/misc.py]: why not just `~scope`???

Comment (yarikoptic) [inline datalad/crawler/nodes/misc.py]: actually here you don't even need a loop.... think! ;)

File: datalad/crawler/nodes/misc.py [hunk 5]
```diff
@@ -29,6 +31,71 @@
...
+                # make is executable for those that can read it
+                for i, scope in enumerate(permissions):
+                    if per[i] == '6' or '4':
+                        os.chmod(filename, st.st_mode | scope)
+                        st = os.stat(filename)
+
+            else:
+                # strip everyone away from executing
+                current = stat.S_IMODE(os.lstat(filename).st_mode)
+                os.chmod(filename, current & ~S_IEXEC & ~S_IXGRP & ~S_IXOTH)
```
Comment (yarikoptic) [inline datalad/crawler/nodes/misc.py]: great ;) or `~(S_IEXEC | S_IXGRP | S_IXOTH)` -- any is good.

**Pull request merged**